### PR TITLE
Add remaining e2e tests

### LINKED ISSUE #18: first draft e2e tests round 2
- [x] no votes on a dispute
- [x] multiple vote rounds on a dispute, all passing
- [x] multiple vote rounds on a dispute, overturn result
- [x] multiple votes from all stakeholders (tokenholders, reporters, users, teamMultisig) (test the handling of edge cases in voting rounds (e.g., voting ties, uneven votes, partial participation))
- [x] test submitting bad identical values (same value, query id, & submission timestamp) on different parachains, disputes open for all, ensure no cross contamination in gov/staking contracts

### PR DESCRIPTION
**Pull request by oraclown**
- closes #18

### CODE REVIEW

File: test/testE2E.t.sol [hunk 1]
```diff
@@ -878,4 +878,79 @@ contract E2ETests is Test {
...
+        (, uint256 _bobStakedBal, uint256 _bobLockedBal,,,,,,) = staking.getParachainStakerInfo(fakeParaId, bob);
+        _bobBalance += _bobStakedBal + _bobLockedBal;
+        assertEq(_bobBalance, _bobStartBalance - fakeSlashAmount);
+        address marge = address(0xbeef);
+        token.mint(address(marge), _bobBalance);
+        uint256 _margeBalance = token.balanceOf(address(marge));
+        assertEq(_bobBalance, _margeBalance);
+        // reporter votes against the dispute
+        vm.prank(bob);
+        gov.vote(_disputeId, false, true);
```
Comment (tkernell) [inline test/testE2E.t.sol:918]: This is more of a comment on the gov contract, but why did we switch the `vote` function's third arg from `invalidQuery` to `validDispute`?

Comment (tkernell) [reply test/testE2E.t.sol:918]: It could just be confusing for people who interact with tellor on both eth mainnet and on polkadot

**Review by tkernell — APPROVED** [review]

File: test/testE2E.t.sol (hunk 1 above)
Comment (oraclown) [reply test/testE2E.t.sol:918]: It might be confusing for a select few, but I doubt it. Not many disputes happen and not many people vote. Also, the argument has the same functionality, I just think it's a more accurate name. `* @param _invalidQuery is whether or not the dispute is valid` vs ` * @param _validDispute is whether or not the dispute is valid (e.g. false if the dispute is invalid)`
